### ISSUE TITLE
[ICC] - Lich King - Teleporting Valkyries/Randomly Speeding/Incorrect Despawn

### ISSUE THREAD

**Issue by ashido96** [dalaranwow/dalaran-wow#4753]
**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
When you stun a valkyrie that has been stunned the valkyrie will instantly teleport. We believe at this time that it is teleporting to the location at which it would be if had not been slowed - we think this to be the case as this bug does not display itself if the valkyrie is stunned within approximately 3 seconds after it becomes target able. 

**Expected behaviour**: Tell us what should happen instead
VALKYRIES SHOULDN'T BLOODY FUCKING TELEPORT WHEN YOU TRY TO CC THEM

http://www.wowhead.com/npc=36609/valkyr-shadowguard
http://www.wowhead.com/npc=36597/the-lich-king

Also since it wont show up in the pulse menu when i try to comment there, open #4730 because it's not fixed.

**Comment by Winstrol** [comment]
Looks like its a core issue with the server  #1630 #1447 #3030

Another issue on heroic is that valks are not tauntable. source: https://youtu.be/h1x9MErf5NA

Also valks are speeding up randomly even with slow applied to them making them sometimes impossible to kill (hc)

**Comment by alexuktc** [comment]
Experienced this as well in 10m normal. It was rather unsettling to stun it in one location and have it tp to another while stunned.

**Comment by plznoz** [comment]
https://youtu.be/8IfY_g-pjo4?t=6m57s
Completely slowed the full duration on live, on Dalaran they speed up like Winstrol says even with multiple slows active.
Maybe lesser slows overwrite the more powerful ones?

**Comment by Crispybuilt** [comment]
Can confirm that Val'kyrs randomly speed up for no reason with the slows being applied. 
Should be a prio to fix asap.
@Winstrol

**Comment by Crispybuilt** [comment]
**Steps to reproduce the problem speeding problem:**
- Let the Val'kyr grab someone
- Apply a random slow effect (Hamstring/Piercing Howl)
- Look at the Val'kyr being slowed, but then speed up to its regular speed for no apparent reason even with multiple slows being applied at the same time

**Steps to reproduce the teleporting problem:**
- Slow the Val'kyr
- Stun it after some time
- Look at it teleport infront a couple of yards (wrong location)

**Comment by Crispybuilt** [comment]
- Updated all the issues with Val'kyr Shadowguard on The Lich King encounter above.

**Comment by Chenocide** [comment]
Bump.

**Comment by Zarjofgt** [comment]
How is this not priority - critical again?

**Comment by Lothloryen** [comment]
Hello there,

Implement a system for the Val'kyr proper management when entering the second transition. Val'kyrs carrying players will not despawn and should work fine on every possible situation - will be on Algalon after the next scheduled maintenance.

I will correct the original issue reported here once I find a proper solution for it, it's being a bit tricky.

Thank you for the report.

**Comment by datbunneh** [comment]
@Lothloryen Is #4986 and issue that caused it fixed in scope of this one?

**Comment by JohnADC** [comment]
Not sure if any part of this is fixed yet but i can confirm valkyrs are still speeding up after slows that last for a certain amount of time(like icy chains form dks)/teleporting after getting stunned(this seems to happen only after a few seconds have already passed and not if stunned immediatly after landing). Will try to get some video recordings of this happening this week on lich king.

**Comment by punims** [comment]
Gonna bump this, because this bug actually is driving the raiders on this server crazy.

**Comment by Chenocide** [comment]
I'm gonna list the workarounds that seemed to work for my guild. 

1: Assign stuns for each Valkyr and only stun them 2 times. DBM should give them marks, so assign stuns based on marks, 1 main stun and 1 backup stun. Seems to be a bad idea to stun a third time cause you might risk that it teleports over the edge. 

2: Only ever assign 1 slow. Applying more than 1 slow seems to make the Valkyrs speed up to 100% speed at least. For example start with a hunter trap and then have a warrior use Piercing Howl and drop a slow totem, any combination will work as long as there is only 1 slow applied at a time.

**Comment by StheNoble** [comment]
this issue is indeed still present and driving us nuts, especially on LoD attempts.

Valks also teleport off the edge and drop the player there when braught to 50% very close to the edge instead of dropping the player on its current position.

i would love a priorization on that because this is the worst bug on that fight right now, besides the unbearable serverlag on sunday evenings...

**Comment by Lothloryen** [comment]
Hello there,

@Rushor  Will take a look to the teleport issue in few days, some fresh eyes might have some enlightment about it.

**Comment by Chenocide** [comment]
@Rushor @Lothloryen What exactly was fixed here, both the speeding issue and the teleporting issue or just one of them? This honestly needs to reach live as fast as possible.

**Comment by Rushor** [comment]
Both - and both will be moved to PTR for testing.

**Comment by StheNoble** [comment]
may i ask whats holding this back from beeing deployed to ptr and beeing tested @Lothloryen ?

**Comment by Zarjofgt** [comment]
How can you let the last fight of the last raid be broken beyond belief? Honest question

**Comment by Dibbarra** [comment]
must be some pretty serious testing for it take this long

**Comment by Dibbarra** [comment]
Is this getting pushed to live soon?

**Comment by Lothloryen** [comment]
Hello there,

Unfortunately the mentioned fixes didn't work. I have applied a correction that, after uncountables tests seems to be working. It will be applied to the live server with tomorrow's maintenance.

Thank you.

**Comment by Lothloryen** [comment]
Hello again,

I couldn't deploy this fix today, it should be done with one of the next maintenances, I will update here when it is up.

**Comment by Lothloryen** [comment]
The fix is up now.
